docs/protocol/specifications/data.md

```diff
@@ -499,14 +505,12 @@ Root hash of the [merkle binary hash tree](#merkle-binary-tree) formed by the *t
 
 Root hash of the [merkle patricia tree](#merkle-patricia-tree) formed by unspent outputs with an **asset version 1** after applying the block. Allows bootstrapping nodes from recent blocks and an archived copy of the corresponding merkle patricia tree without processing all historical transactions.
 
-The tree contains [non-retired](#retired-asset) unspent outputs (one or more per [asset ID](#asset-id)):
+The tree contains [non-retired](#retired-asset) unspent outputs (one or more per [asset ID](#asset-id)) where both key and value are the same value — the [Unspent ID](#unspent-id) of the unspent output.
```

Comment:
We need to define how to compute the merkle root now to not break hashes. But to actually make it optimizeable into a set using _the current merkle patricia trie_ algorithm, we need to define value to be equal to a key. This achieves these things:

1. Our patricia trie implementation does not need to be changed now.
2. Leafs are more strongly committed to keys than if they were empty values and the resulting hash becomes position-dependent. Feels more robust than having empty values to me.
3. Nice side effect: our existing patricia trie algorithm is more versatile - can be used both for hashing key-values and sets as well.

WDYT on the point 2? @bobg @kr